Utils.py

```diff
@@ -468,6 +468,18 @@ def restricted_loads(s: bytes) -> Any:
     return RestrictedUnpickler(io.BytesIO(s)).load()
 
 
+def restricted_dumps(obj: Any) -> bytes:
+    """Helper function analogous to pickle.dumps()."""
+    s = pickle.dumps(obj)
+    # Assert that the string can be successfully loaded by restricted_loads
+    try:
+        restricted_loads(s)
+    except pickle.UnpicklingError as e:
+        raise pickle.PicklingError(e)
```

Comment:
I like this part of Vi's comment
> errors with "the pickle was successful, but this data cannot be unpickled with RestrictedUnpickler"

https://discord.com/channels/731205301247803413/1214608557077700720/1383399333902155887

about making it clear that the pickling succeeded, but the unpickling failed.

This seems to do the opposite, catching `UnpicklingError` to raise `PicklingError`, making it  possibly confusing and less clear.

I would like it to be clear that serialization succeeded, but the test of deserialization failed.

Reply:
Yeah adding an extra line to that message would be good explaining what it means

Reply:
I thought about that, but really the error here is the same whether it happens on the pickling or unpickling side, and to me the fact that it gets caught since it's attempting to unpickle rather than just checking it directly is more of an implementation detail that's probably not relevant to a caller when it fails.
If it's desired though, I'd be fine with be adding a prefix like `PicklingError: failed to unpickle with error:` or something like that. The `global '{module}.{name}' is forbidden` message on the unpickler itself is also maybe not the most clear, so that could be improved itself.

Reply:
That makes sense to me

Reply:
Yeah it's ok.
```suggestion
    try:
        restricted_loads(s)
    except pickle.UnpicklingError as e:
        raise pickle.PicklingError(e) from e
```
This adjusts the error message a little to make it look less like multiple errors unexpectedly compounding.